tools/platforms/MacPlatform.hx

```diff
@@ -95,22 +95,6 @@ class MacPlatform extends PlatformTarget
 				title: ""
 			};
 
-		switch (System.hostArchitecture)
-		{
-			case ARMV6:
-				defaults.architectures = [ARMV6];
-			case ARMV7:
-				defaults.architectures = [ARMV7];
-			case X86:
-				defaults.architectures = [X86];
-			case X64:
-				defaults.architectures = [X64];
-			case ARM64:
-				defaults.architectures = [ARM64];
-			default:
-				defaults.architectures = [];
-		}
-
```

Comment:
I'm not sure if I'm missing something, but if this switch case is removed, won't `project.architectures` just always be empty? @player-03

Reply:
Ah, I guess it doesn't matter because `project.architectures` is not used anymore past this point?

Reply:
`project.architecture` is used on line 114: `for (architecture in project.architectures)`.

The key is that it may be set by the user, in project.xml. The removed code was setting the default architecture in case the user didn't set any. My new code sets the default on line 113: `targetArchitecture = Type.createEnum(...)`. Once set, it reads from `project.architecture` in case the user overrode it.

The main difference is that in the past it would ignore any user-provided value other than x64, but now the user can provide one of five values. If the user provides no value, it will still default to the host architecture.

Reply:
Not that most of the options have any meaning. As before, it has a special case for the 64-bit options, and otherwise passes no extra arguments to hxcpp. So even if the user set x86 on an Arm machine, we would run hxcpp with no x86-specific arguments, and hxcpp would presumably build for Arm.

Reply:
Thanks for the explanation, that makes sense!